packages/@aws-cdk/cdk/test/test.duration.ts

```diff
@@ -0,0 +1,108 @@
+import nodeunit = require('nodeunit');
+import { Duration, Stack, Token } from '../lib';
+
+export = nodeunit.testCase({
+  'negative amount'(test: nodeunit.Test) {
+    test.throws(() => Duration.seconds(-1), /negative/);
+
+    test.done();
+  },
+
+  'unresolved amount'(test: nodeunit.Test) {
```

Comment:
Mmmm.... I wonder... If no unit conversion is needed, there shouldn't be a problem supporting tokens, and it's actually something we should support:

```ts
Duration.seconds(durationParameter.valueAsNumber).toSeconds()
```

This should work...